### DISCUSSION TITLE
Decimals

### DISCUSSION THREAD

**Discussion by NevileeV** [BG-Software-LLC/SuperiorSkyblock2#1716] [Pending Feature Requests] [answered]
Pls add decimals to your plugin, when i se in generator-rats
Example:
    '2':
      price: 1500.0
      price-type: 'placeholders'
      placeholder: '%gamepoints_balance%'
      withdraw-commands:
      - 'gp take %player% 1500'
      generator-rates:
        normal:
          STONE: 92
          COAL_ORE: 5
          REDSTONE_ORE: 1
          IRON_ORE: 2
          GOLD_ORE: 0.5
      commands:
        - 'island admin setupgrade %player% generator-rates 3'
        - 'island admin msgall %player% &e&lUlepszenie | &7Generator został ulepszony!'

gold ore didnt spawn :/ ... when i set 1 all work

**Comment by OmerBenGera** [comment, marked as answer]
You can have decimal percentage by multiplying everything by 10.
The way the percentage of the rates is calculated is by taking the value of the ore and divide it by the sum of all the rates together.
Therefore, if you have the following setup:
```
STONE: 999
COAL_ORE: 1
```
There will be a 99.9% chance for stone to generate, while only 0.1% for coal ore.
